### PR TITLE
WT-11150 Cast involving volatile to no effect in assertion

### PR DESCRIPTION
**Pull request by ershov**

### CODE REVIEW

File: src/include/btree_inline.h [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ __wt_ref_addr_copy(WT_SESSION_IMPL *session, WT_REF *ref, WT_ADDR_COPY *copy)
          * page with rec_result == 0 we will free the addr being copied - possibly after the null
          * check above - and this function will attempt to copy from freed memory.
          */
-        WT_ASSERT(session, (volatile void *)ref->addr != NULL);
+        WT_ASSERT(session, (void * volatile)ref->addr != NULL);
```
Comment (korteland) [inline src/include/btree_inline.h:1562]: This doesn't match up with the intent of the ticket - the type here should be `(void * volatile *)&ref->addr, with a suitable dereference out the front. The idea is that we don't care about the memory that `ref->addr` is pointing to - rather, we need to treat `addr` itself as volatile.

Comment (ershov) [reply src/include/btree_inline.h]: Done.

**Review by marcbutler — APPROVED** [review]
LGTM

**Review by korteland — APPROVED** [review]
LGTM - congrats on your first WiredTiger PR!

**Pull request merged**